apps/server/src/modules/user-login-migration/service/user-login-migration.service.ts

```diff
@@ -47,15 +47,14 @@ export class UserLoginMigrationService {
 		return updatedUserLoginMigration;
 	}
 
-	public async setMigrationMandatory(schoolId: string, mandatory: boolean): Promise<UserLoginMigrationDO> {
-		// this.checkGracePeriod(userLoginMigration);
-		let userLoginMigration: UserLoginMigrationDO | null = await this.userLoginMigrationRepo.findBySchoolId(schoolId);
+	public async setMigrationMandatory(userLoginMigration: UserLoginMigrationDO): Promise<UserLoginMigrationDO> {
+		this.checkGracePeriod(userLoginMigration);
 
-		if (!userLoginMigration) {
-			throw new UserLoginMigrationNotFoundLoggableException(schoolId);
+		if (userLoginMigration.closedAt) {
+			throw new UserLoginMigrationAlreadyClosedLoggableException(userLoginMigration.closedAt, userLoginMigration.id);
 		}
 
-		if (mandatory) {
+		if (userLoginMigration.mandatorySince) {
```

Comment:
this is wrong. mandatory is an input of the function and describes the requested state. userLoginMigration.mandatorySince describes the current state. (that should or should not be changed)

Reply:
Thanks!